src/mqueryfront/src/QueryField.js

```diff
@@ -1,93 +1,12 @@
 import React, { Component } from "react";
-import axios from "axios";
-import { API_URL } from "./config";
 import QueryMonaco from "./QueryMonaco";
 
 class QueryField extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            selectedTaint: null,
-            error: null,
-        };
-
-        this.handleInputChange = this.handleInputChange.bind(this);
-        this.handleYaraChanged = this.handleYaraChanged.bind(this);
-        this.handleQuery = this.handleQuery.bind(this);
-        this.handleEdit = this.handleEdit.bind(this);
-    }
-
-    componentDidMount() {}
-
-    selectTaint(newTaint) {
-        this.setState({
-            selectedTaint: newTaint,
-        });
-    }
-
     describeTaint() {
-        if (this.state.selectedTaint == null) {
+        if (this.props.selectedTaint == null) {
```

Comment:
maybe `this.props.selectedTaint or "everywhere"` (I know it was originally written by me, but sounds like an improvement - thoughts?)

Reply:
Sure, looks like good case for short circuiting!